tests/Integration/Behaviour/Features/Context/Domain/Product/Combination/GenerateCombinationFeatureContext.php

```diff
@@ -43,18 +43,48 @@ class GenerateCombinationFeatureContext extends AbstractCombinationFeatureContex
      * @param string $productReference
      * @param TableNode $table
      */
-    public function generateCombinations(string $productReference, TableNode $table): void
+    public function generateCombinationsForDefaultShop(string $productReference, TableNode $table): void
     {
-        $tableData = $table->getRowsHash();
-        $groupedAttributeIds = $this->parseGroupedAttributeIds($tableData);
+        $this->generateCombinations($productReference, $table, $this->getDefaultShopId());
+    }
 
-        try {
-            $this->getCommandBus()->handle(new GenerateProductCombinationsCommand(
-                $this->getSharedStorage()->get($productReference),
-                $groupedAttributeIds
-            ));
-        } catch (InvalidProductTypeException $e) {
-            $this->setLastException($e);
+    /**
+     * @When I generate combinations in shop ":shopReference" for product :productReference using following attributes:
+     *
+     * @param string $shopReference
+     * @param string $productReference
+     * @param TableNode $table
+     */
+    public function generateCombinationsForShop(string $shopReference, string $productReference, TableNode $table): void
```

Comment:
yes, please dedicated PR :smile:

Reply:
Since you already handled the use case in the generation command I still think this should be tested in this PR as well You already handled it all in the command so it's just adding one scenario to validate it

And that's why I think you'll need a `Combination product1SWhite is associated to "shop1,shop2"` which allows checking multiple shop associations with a single reference

Reply:
Ok to handle this in the following PR, then we'll wait for it to be over and integrated to ask for QA on this feature But it needs to be done quickly after this PR is merged while it's still fresh in our minds